### ISSUE TITLE
Issues in the generation step

### ISSUE THREAD

**Issue by frallebini** [POZAlabs/ComMU-code#11]
Hi,

I ran both the preprocessing and the training as described in the README (after your last commit). However, when I try to generate a sample with the metadata reported as an example in the README, namely
```
--bpm 70 \
--audio_key aminor \
--time_signature 4/4 \
--pitch_range mid_high \
--num_measures 8 \
--inst acoustic_piano \
--genre newage \
--min_velocity 60 \
--max_velocity 80 \
--track_role main_melody \
--rhythm standard \
--chord_progression Am-Am-Am-Am-Am-Am-Am-Am-G-G-G-G-G-G-G-G-F-F-F-F-F-F-F-F-E-E-E-E-E-E-E-E-Am-Am-Am-Am-Am-Am-Am-Am-G-G-G-G-G-G-G-G-F-F-F-F-F-F-F-F-E-E-E-E-E-E-E-E \
--num_generate 3
```
I either get (when using `checkpoint_best.pt`)
```
UserWarning: Setting attributes on ParameterList is not supported.
2022-11-20 15:13:37,406 | INFO | ComMU | Generating the idx: 1
2022-11-20 15:13:39,677 | INFO | ComMU | correct_length: 8
2022-11-20 15:13:39,677 | INFO | ComMU | <list of indexes>
2022-11-20 15:13:39,677 | ERROR | ComMU | Empty sequence generated
2022-11-20 15:13:39,678 | INFO | ComMU | Generating the idx: 1
2022-11-20 15:13:41,024 | INFO | ComMU | correct_length: 8
2022-11-20 15:13:41,024 | INFO | ComMU | <list of indexes>
2022-11-20 15:13:41,025 | ERROR | ComMU | Empty sequence generated
2022-11-20 15:13:41,025 | INFO | ComMU | Generating the idx: 1
2022-11-20 15:13:42,897 | INFO | ComMU | correct_length: 8
2022-11-20 15:13:42,897 | INFO | ComMU | <list of indexes>
```
which loops forever, or (when using `checkpoint_last.pt`)
```
UserWarning: Setting attributes on ParameterList is not supported.
2022-11-20 15:30:45,402 | INFO | ComMU | Generating the idx: 1
2022-11-20 15:31:05,013 | ERROR | ComMU | remnant chord length: 8 
error in teacher forcing
2022-11-20 15:31:05,013 | INFO | ComMU | Generating the idx: 1
2022-11-20 15:31:24,684 | ERROR | ComMU | remnant chord length: 8 
error in teacher forcing
```
which also loops forever.

What is the issue? The only change that I made is a reduction in the batch size to fit the data in my (single) GPU. 

Alternatively, Is there any way you could give me a checkpoint like the one you used to generate the demos in the project page?

**Comment by hyeonch** [comment]
see config helper and fix batch chunk and batch size. if you used the script we gave without fixing nproc_per_node, you have to fix it, too
the code was written for using DDP, and in case DDP won't work in your environment, you have to fix code executable without DDP

**Comment by frallebini** [comment]
I trained with `--nproc_per_node=1`. What do you mean by "fix batch chunk"?. If `batch_chunk` (4) is still a divisor of `batch_size` (64), shouldn't that be ok?

**Comment by hyeonch** [comment]
set batch_chunk as 1 in your case, and please inspect the code and fix other configs that depends on the number of devices such as learning rate.

**Comment by frallebini** [comment]
Does `batch_chunk` necessarily have to be the same as `--nproc_per_node`?

**Comment by frallebini** [comment]
I trained the model following your suggestions, and increased the number of training steps (up to 160,000) in an attempt to compensate for the smaller batch size (now 32, since I cannot go higher with `batch_chunk` set to 1).

When I run the generation scripts, the model does not get stuck in infinite loops anymore, but the generated sample are far from satisfying: they are mostly silence with one or two bars. It seems clear to me at this point that I do not have the computational budget necessary to reproduce your results.

However, I will ask again: since my research interest is mainly to focus on what you call "stage 2" of the combinatorial music generation problem, and not on the model training itself, would it be possible to have one of your checkpoints? My institutional mail is francesco.ballerini3@studio.unibo.it.

**Comment by hyeonch** [comment]
@frallebini sorry for late answer to your inquiry. since whole team in New Orleans for the conference, it's difficult for us to meet your needs right about now. we will upload checkpoint file we used after the conference, so please wait for about a week. Thank you.
